### PR TITLE
refactor: cleanup colors

### PR DESCRIPTION
**Pull request by markov00**
## Summary

During the heatmap legend fixes I've noticed that we have multiple places where some standard colors are allocated.
I've moved the `Color` type + some static colors (red, white, black, transparent) into the `common/color.ts` file.
This is not a complete refactoring but the first part of the initiative to cleanup the folders the typings.

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)

### CODE REVIEW

File: packages/charts/src/utils/themes/theme_common.ts [hunk 1]
```diff
@@ -6,28 +6,29 @@
...
  */
 
+import { Color, Colors } from '../../common/colors';
 import { Margins } from '../dimensions';
 import { SharedGeometryStateStyle } from './theme';
 
-/** @public */
-export const DEFAULT_MISSING_COLOR = 'red';
+/** @internal */
+export const DEFAULT_MISSING_COLOR: Color = Colors.Red.keyword;
```
Comment (nickofthyme) [inline packages/charts/src/utils/themes/theme_common.ts:14]: the `Color.X.keyword` should already be typed as `Color`.

```suggestion
export const DEFAULT_MISSING_COLOR = Colors.Red.keyword;
```

File: packages/charts/src/chart_types/goal_chart/renderer/canvas/connected_component.tsx [hunk 2]
```diff
@@ -178,7 +178,7 @@ const DEFAULT_PROPS: ReactiveChartStateProps = {
   bandLabels: [],
   firstValue: 0,
   captureBoundingBox: initialBoundingBox(),
-  background: 'transparent',
+  background: Colors.Transparent.keyword,
```
Comment (rshen91) [inline packages/charts/src/chart_types/goal_chart/renderer/canvas/connected_component.tsx:181]: love this change!

**Review by nickofthyme — APPROVED** [review]
Code changes LGTM.

**Review by rshen91 — APPROVED** [review]

File: packages/charts/src/utils/themes/theme_common.ts (hunk 1 above)
Comment (markov00) [reply packages/charts/src/utils/themes/theme_common.ts]: done

**Pull request merged**